lib/bitcoin/BitcoinClient.ts

```diff
@@ -162,15 +214,81 @@ export default class BitcoinClient implements IBitcoinClient {
   }
 
   // This function is specifically created to help with unit testing.
-  private static createTransactionFromBuffer (buffer: Buffer): Transaction {
+  private static createBcoreTransactionFromBuffer (buffer: Buffer): Transaction {
     return new Transaction(buffer);
   }
 
   // This function is specifically created to help with unit testing.
-  private static createBlockFromBuffer (buffer: Buffer): Block {
+  private static createBcoreBlockFromBuffer (buffer: Buffer): Block {
     return new Block(buffer);
   }
 
+  private async createBcoreTransaction (transactionData: string, feeInSatoshis: number): Promise<Transaction> {

```

Comment:
Consider changing the method to `createRawTransaction()` and return `transaction.serialize()` instead. This way Bitcore specific logic is not exposed to caller unnecessarily.